src/global/trampoline.rs

```diff
@@ -1,66 +1,299 @@
-const TRAMPOLINE_BIN: &[u8] = include_bytes!(
+pub const TRAMPOLINE_BIN: &[u8] = include_bytes!(
     "../../crates/pixi_trampoline/trampolines/pixi-trampoline-x86_64-unknown-linux-musl"
 );
 
+// return file name of the executable
+pub(crate) fn file_name(exposed_name: &ExposedName) -> String {
+    if cfg!(target_os = "windows") {
+        format!("{}.exe", exposed_name)
+    } else {
+        exposed_name.to_string()
+    }
+}
+
+/// Extracts the executable path from a script file.
+///
+/// This function reads the content of the script file and attempts to extract
+/// the path of the executable it references. It is used to determine
+/// the actual binary path from a wrapper script.
+pub(crate) async fn extract_executable_from_script(script: &Path) -> miette::Result<PathBuf> {
+    // Read the script file into a string
+    let script_content = tokio_fs::read_to_string(script).await.into_diagnostic()?;
+
+    // Compile the regex pattern
+    #[cfg(unix)]
+    const PATTERN: &str = r#""([^"]+)" "\$@""#;
+    // The pattern includes `"?` to also find old pixi global installations.
+    #[cfg(windows)]
+    const PATTERN: &str = r#"@"?([^"]+)"? %/*"#;
+    static RE: Lazy<Regex> = Lazy::new(|| Regex::new(PATTERN).expect("Failed to compile regex"));
+
+    // Apply the regex to the script content
+    if let Some(caps) = RE.captures(&script_content) {
+        if let Some(matched) = caps.get(1) {
+            return Ok(PathBuf::from(matched.as_str()));
+        }
+    }
+    tracing::debug!(
+        "Failed to extract executable path from script {}",
+        script_content
+    );
+
+    // Return an error if the executable path couldn't be extracted
+    miette::bail!(
+        "Failed to extract executable path from script {}",
+        script.display()
+    )
+}
+
+/// Manifest data of the original executable.
+/// This is used by trampoline to set the environment variables
+/// prepened the path and execute the original executable.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct ManifestMetadata {
+    /// Path to the original executable.
+    pub exe: PathBuf,
+    /// Root path of the original executable that should be prepended to the PATH.
+    pub path: PathBuf,
+    /// Environment variables to be set before executing the original executable.
+    pub env: HashMap<String, String>,
+}
+
+impl ManifestMetadata {
+    /// Create a new manifest metadata.
+    pub fn new(exe: PathBuf, path: PathBuf, env: Option<HashMap<String, String>>) -> Self {
+        ManifestMetadata {
+            exe,
+            path,
+            env: env.unwrap_or_default(),
+        }
+    }
+
+    /// Read existing manifest metadata from the root path.
+    pub async fn from_root_path(
+        root_path: PathBuf,
+        exposed_name: &ExposedName,
+    ) -> miette::Result<Self> {
+        let manifest_path = root_path.join(exposed_name.to_string() + ".json");
+        let manifest_str = tokio::fs::read_to_string(manifest_path)
```

Comment:
```suggestion
        let manifest_str = tokio_fs::read_to_string(manifest_path)
```
Please convert all of these